### PR TITLE
[PP] Allow larger world_size schedule tests

### PR DESCRIPTION
**Pull request by H-Huang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #160474
* __->__ #160559
* #160558

Update schedule tests to use `world_size=4`, changes needed:
- Move some tests that require world_size=2 to new class
- Move helper methods from class level to function level
- Update some initialization to pass assert since gradients were super small.

cc @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @pragupta

### CODE REVIEW

File: test/distributed/pipelining/model_registry.py [hunk 1]
```diff
@@ -211,10 +211,10 @@ def __init__(self, d_hid: int):
         self.fc2_weight = torch.nn.Parameter(torch.randn(d_hid, d_hid))
         self.fc2_bias = torch.nn.Parameter(torch.randn(d_hid))
 
-        torch.nn.init.uniform_(self.fc1_weight, -0.01, 0.01)
-        torch.nn.init.uniform_(self.fc2_weight, -0.01, 0.01)
-        torch.nn.init.uniform_(self.fc1_bias, -0.01, 0.01)
-        torch.nn.init.uniform_(self.fc2_bias, -0.01, 0.01)
+        torch.nn.init.uniform_(self.fc1_weight, -0.001, 0.001)
```
Comment (wconstab) [inline test/distributed/pipelining/model_registry.py:214]: ?

**Review by wconstab — APPROVED** [review]

File: test/distributed/pipelining/model_registry.py (hunk 1 above)
Comment (H-Huang) [reply test/distributed/pipelining/model_registry.py:214]: In only one of the tests we use this and the module weight initialization effects the scale of the gradients.

When I changed the test from 2->4 ranks, the gradients became super small like 1e-10 due to this initialization, making these weights smaller helps fix that.
